src/compas_timber/elements/fasteners/plate_fastener.py

```diff
@@ -0,0 +1,319 @@
+        The holes of the fastener. Structure is as follows: [(point, diameter), ...]
+    angle : float, optional (default=math.pi / 2)
+        The angle of the fastener. The angle between the beam elements must be the same.
+
+    Attributes
+    ----------
+    geometry : :class:`~compas.geometry.Geometry`
+        The geometry of the fastener.
+    frame : :class:`~compas.geometry.Frame`
+        The frame of the fastener.
+
+    """
+
+    def __init__(
+        self,
+        outline=None,
+        thickness=None,
+        interfaces=None,
+        frame=None,
+        angle=math.pi / 2,
+        topology=None,
+        cutouts=None,
+        **kwargs
+    ):
+        super(PlateFastener, self).__init__(**kwargs)
+        self.outline = outline
+        self.thickness = thickness
+        self.interfaces = interfaces
+        self.frame = frame or Frame.worldXY()
+        self.angle = angle
+        self.topology = topology
+        self.cutouts = cutouts
+        self.attributes = {}
+        self.attributes.update(kwargs)
+        self.debug_info = []
+
+    @property
+    def __data__(self):
+        data = super(PlateFastener, self).__data__
+        data["outline"] = self.outline
+        data["thickness"] = self.thickness
+        data["interfaces"] = [interface.__data__ for interface in self.interfaces]
+        data["frame"] = self.frame
+        data["angle"] = self.angle
+        data["topology"] = self.topology
+        data["cutouts"] = self.cutouts
+        return data
+
+    @classmethod
+    def __from_data__(cls, data):
+        return cls(
+            outline=data["outline"],
+            thickness=data["thickness"],
+            interfaces=[FastenerTimberInterface.__from_data__(interface) for interface in data["interfaces"]],
+            frame=data["frame"],
+            angle=data["angle"],
+            topology=data["topology"],
+            cutouts=data["cutouts"],
+        )
+
+    def __repr__(self):
+        # type: () -> str
+        return "Plate Fastener(frame={!r}, name={})".format(self.frame, self.name)
+
+    def __str__(self):
+        # type: () -> str
+        return "<Plate Fastener {} at frame={!r}>".format(self.name, self.frame)
+
+    @property
+    def frame(self):
+        return self._frame
+
+    @frame.setter
+    def frame(self, frame):
+        self._frame = frame
+        for interface in self.interfaces:
+            if interface is not None:
+                interface.frame = frame
+
+    @property
+    def holes(self):
+        for interface in self.interfaces:
+            for hole in interface.holes:
+                yield hole
+
+    @property
+    def shapes(self):
+        for interface in self.interfaces:
+            for shape in interface.shapes:
+                yield shape
+
+    def set_default(self, joint):
+        width_a = joint.beams[0].width
+        width_b = joint.beams[1].width
+        if joint.SUPPORTED_TOPOLOGY == 3:  # JointTopology.TOPO_T
```

Comment:
notice `SUPPORTED_TOPOLOGY` is a list now.
also if possible, use `JointTopology.TOPO_` instead of the actual value

Reply:
yeah I was getting circular import errors when importing `joint.SUPPORTED_TOPOLOGY` so I went with the `int` value. 